docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -135,25 +139,54 @@ def get(self, request, projectid, filename):
 
     # TODO refactor this function by moving the actual upload and Project model updates to library function outside the view
     def post(self, request, projectid, filename, format=None):
-        project = Project.objects.get(id=projectid)
+        """Handle POST requests."""
+
+        # FIXME: CURRENTLY UNDERGOING DEBUG
+        # Write traceback to buffer + deepcopy request before anything happens to the request object
+        buffer = io.StringIO()
+        print_stack(limit=50, file=buffer)
+
+        # Fetch important data without mutating request
+        request_attributes = {
+            "data": str(copy.copy(request.data).keys()),
+            "files": str(request.FILES.keys()),
+            "meta": str(request.META),
+        }
+
+        if len(request.FILES.getlist("file")) > 1:
+            raise exceptions.MultipleContentsError()
+
+        missing_error = ""
 
         if "file" not in request.data:
-            logger.info(
-                'The key "file" was not found in `request.data`.',
-                extra={
-                    "data_for_key_text_content": str(request.data.get("text", ""))[
-                        :1000
-                    ],
-                    "data_for_key_text_len": len(request.data.get("text", "")),
-                    "request_content_length": request.META.get("CONTENT_LENGTH"),
-                    "request_content_type": request.META.get("CONTENT_TYPE"),
-                    "request_data": list(request.data.keys()),
-                    "request_files": list(request.FILES.keys()),
-                },
-            )
+            missing_error = 'The key "file" was not found in `request.data`. Sending report to Sentry.'
+
+        if not request.FILES.getlist("file"):
+            missing_error = 'The key "file" occurs in `request.data` but maps to an empty list. Sending report to Sentry.'
+
+        # When missing_error is truthy...
+        if missing_error:
+            logging.warning(missing_error)
+
+            # ...report to sentry if the length is below the 10MBs
+            if int(request.META.get("CONTENT_LENGTH")) < 10000000:
```

Comment:
Can we also report but without attachements (or trimmed attachments) otherwise?

Reply:
The attachments are just strings now so it should be light-footed enough.